.github/workflows/docker-image.yml

```diff
@@ -2,7 +2,9 @@ name: Docker Image CI
 
 on:
   push:
-    branches: ["**"]
```

Comment:
TLDR; I'm ok with only building release and master builds on every commit because in practice the other random branches tend to fail tests and we run those locally.

Currently, the last step `push` only pushes a docker image if:

```
push: ${{ startsWith(github.ref, 'refs/heads/release/') || startsWith(github.ref, 'refs/tags/') }}
```

So dev branches already don't push to docker hub.  I wanted all branches to build because I liked the idea of pushing any branch and having the GitHub actions build and test it without necessarily pushing.  In practice however, most of the time these tests fail because of acasclient and other repo changes which are not reflected in the current default branch builds (which the build falls back to if it can't find tags with the same name as the current branch).

Reply:
@brianbolt Thanks for the comment. In this PR I'm adding the new behavior to build & test (but not push) on PR, so I think builds on feature branches will still happen - they'll just only happen once you open a PR, and then on subsequent commits to an open PR.